autoscaler/controllers/actions/probabilisticsampler_controller.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package actions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	actionv1 "github.com/odigos-io/odigos/api/actions/v1alpha1"
+	v1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/common"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var supportedSignals = map[common.ObservabilitySignal]struct{}{
+	common.TracesObservabilitySignal: {},
+}
+
+type ProbabilisticSamplerReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *ProbabilisticSamplerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	logger := log.FromContext(ctx)
+	logger.V(0).Info("Reconciling ProbabilisticSampler action")
+
+	action := &actionv1.ProbabilisticSampler{}
+	err := r.Get(ctx, req.NamespacedName, action)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	for _, signal := range action.Spec.Signals {
+
+		if _, ok := supportedSignals[signal]; !ok {
+
+			err = fmt.Errorf("unsupported signal: %s", signal)
+			logger.V(0).Error(nil, err.Error())
+			r.ReportReconciledToProcessorFailed(ctx, action, FailedToTransformToProcessorReason, err.Error())
+			return ctrl.Result{}, nil
+		}
+	}
+
+	processor, err := r.convertToProcessor(action)
+	if err != nil {
+		logger.V(0).Error(nil, err.Error())
```

Comment:
Can this work as well?

```suggestion
		logger.V(0).Error(err, "failed to convert ProbabilisticSampler to processor")
```

Reply:
Yes, thanks